kits/arm/ex_gravity_compensation_toggle.cpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * This file is a barebones skeleton of how to setup an arm for use.
+ * It demonstrates gravity compensation behavior by commanding torques
+ * equal to the force from gravity on the links and joints of an arm.
+ * Note that this only approximately balances out gravity, as imperfections in
+ * the torque sensing and modeled system can lead to "drift".  Also, the
+ * particular choice of PID control gains can affect the performance of this
+ * demo.
+ */
+
+#include "group_command.hpp"
+#include "group_feedback.hpp"
+#include "robot_model.hpp"
+#include "arm/arm.hpp"
+#include "util/mobile_io.hpp"
+#include <chrono>
+#include <thread>
+#include "hebi_util.hpp"
+
+using namespace hebi;
+using namespace experimental;
+
+int main(int argc, char* argv[])
+{
+  //////////////////////////
+  ///// Config Setup ///////
+  //////////////////////////
+
+  // Config file path
+  const std::string example_config_file = "config/ex_gravity_compensation_toggle.cfg.yaml";
+  std::vector<std::string> errors;
+  
+  // Load the config
+  const auto example_config = RobotConfig::loadConfig(example_config_file, errors);
+  for (const auto& error : errors) {
+    std::cerr << error << std::endl;
+  }
+  if (!example_config) {
+    std::cerr << "Failed to load configuration from: " << example_config_file << std::endl;
+    return -1;
+  }
+
+  // For this demo, we need the arm and mobile_io
+  std::unique_ptr<hebi::experimental::arm::Arm> arm;
+  std::unique_ptr<hebi::util::MobileIO> mobile_io;
+
+  //////////////////////////
+  ///// Arm Setup //////////
+  //////////////////////////
+
+  // Create the arm object from the configuration
+  arm = hebi::experimental::arm::Arm::create(*example_config);
+
+  // Keep retrying if arm not found
+  while (!arm) {
+      std::cerr << "Failed to create arm, retrying..." << std::endl;
+
+      // Wait for 1 second before retrying
+      std::this_thread::sleep_for(std::chrono::seconds(1));  
+
+      // Retry
+      arm = hebi::experimental::arm::Arm::create(*example_config);
+  }
+  std::cout << "Arm connected." << std::endl;
+
+  // Retrieve the gravcomp plugin from the arm
+
+  // Pointer magic
+
+  // Lock the weak_ptr and get a shared_ptr
+  auto plugin_shared_ptr = arm->getPluginByName("gravComp").lock();
+
+  // Check if the shared_ptr is valid
+  if (!plugin_shared_ptr) {
+    std::cerr << "Failed to lock plugin shared_ptr. The plugin may have been destroyed." << std::endl;
+    return -1;
+  }
+
+  // Downcast to ImpedanceController
+  auto gravcomp_plugin_ptr = std::dynamic_pointer_cast<hebi::experimental::arm::plugin::GravityCompensationEffort>(plugin_shared_ptr);
+
+  if (!gravcomp_plugin_ptr) {
+    std::cerr << "Failed to cast plugin to GravityCompensationEffort." << std::endl;
+    return -1;
+  }
```

Comment:
If we can use the `getPluginByType` functions instead, I think this will be more readable.  Less dynamic casting there, and more compile-time checks (right now, we're assuming the plugin is named `gravComp` in the config file, which is a bit of a loose coupling, for example)

Reply:
Done